### ISSUE TITLE
Tags for vegetation

### ISSUE THREAD

**Issue by quaestor** [theCrag/website#2931]
**What happened?**

Sometimes, the aspect of a crag is not enough to decide whether it's suitable for summer climbing (south facing but in the shade due to trees) or winter climbing (south facing with deciduous trees). For the same reason, the tag "all day sun" wouldn't apply in such situations, because it's season-dependent.

**What you expected:**

I suggest a series of tags with icons indicating the height of the surrounding vegetation, e.g. like these (scanned from a local guide book):

![nothing](https://user-images.githubusercontent.com/54371/31840506-f8b8e2a0-b5e5-11e7-9c15-916e8ca64f75.png)
![shrubs](https://user-images.githubusercontent.com/54371/31840507-f8d4f918-b5e5-11e7-8533-913d24457e3b.png)
![trees](https://user-images.githubusercontent.com/54371/31840508-f8eee38c-b5e5-11e7-9e02-c189a217b814.png)

**Comment by brendanheywood** [comment]
The aspect, ie N / S / E / W is already independence from the tags for the sun. Also the sun tags have other issues, and we've considered whether they should not be tags but something more specific, like a range of hours, eg 10am - 3pm. Regardless of whether it's trees causing the shade, or the direction of the cliff, the sunlight could be different between winter and summer either way. At the end of the day you are less concerned with why it's shady, just wether it is or isn't. 

So I'm thinking perhaps the data model could be something like, start / stop sun at exteme winter and extreme summer:

```
Summer sun: [6am] -- [4pm]
Winter sun: [9am] -- [1pm]
```

Thoughts?

**Comment by quaestor** [comment]
I like the idea, especially because it would be great in combination with search queries.

But vegetation is not only relevant for how sunny a crag gets, it may also shelter from wind, or access to the routes could be hindered by brushwood, nettles, etc.

**Comment by lordyavin** [comment]
I don't expect that this fine grained detail will be set unless the a guide book states it already. I fear it is not worth the effort.

**Comment by quaestor** [comment]
I have to agree with @lordyavin, most people won't be so dedicated as to camp near the rock to measure the exact times, and on second thought it's also hard to enter partial information. Enough tags would mitigate this, like additional "winter/cold temperature crag", "summer/hot temperature crag" tags etc.

**Comment by nicHoch** [comment]
i like the vegetation information and use it a lot if the guide books provides it.

**Comment by rouletout** [comment]
My point of view is that it doesn't really add value over the exposition and sun exposure tags - more tags will not mean that they will be used. Stuff like that is often better described in words in the area description, especially if it is big differences over different seasons...

**Comment by quaestor** [comment]
You cannot systematically search for words in the description. And to be honest, I think more climbers would like to know that there are large trees in front of the wall than are interested in the exact geological name of the rock type they are climbing on; and we have even that. Moreover, at least I would certainly use the tag. And extensively so.

I acknowledge the problem of the seasons. But then again: "coniferous trees" as a tag, e.g. for a south face, would make it much clearer that it's sort of shady even in the summer, whereas "deciduous trees" would indicate that it gets some sun in the winter. It is therefore *more* expressive than the "sun exposure" tags.

**Comment by rouletout** [comment]
So this would mean we would need these 2 tags only (coniferous trees, deciduous trees)? There was some talk about the size of trees higher up... I find that even harder to manage as trees tend to grow :wink:

**Comment by quaestor** [comment]
But if you're worried about that, none of the tags might make sense. They all require a certain amount of care. For example, I was at a crag today where the trees got cut down recently. So no matter which set of tags you have, you'll have to adjust them. In my case it was a north face, so not much to be done, but if trees get cut down in a south face, you might have to change it from "always shady" to "all day sun" in the extreme case.

And to the point: I would distinguish between "just grass", "bushes" (in contrast to real trees), and the two types of trees. I would not leave out the first one, because presence of a tag usually says more than absence of one. And for "bushes", this would be just to indicate that there might be something fishy, like a seasonal access problem due to stinging nettles, as already discussed (which of course would have to be put into the description, too). But I'm happy to find a compromise.

I agree that the amount of tags is already quite substantial, and that most people don't bother updating the crags they go to, but I don't think it's because people don't know how to use them. There's just no real incentive to do so because the coverage is too low. I can only speak for myself here, but I am working towards having most of the tags set for the crags in my home area, sometimes putting quite some effort into systematically setting them for whole valleys. And I often find myself in the situation that I don't put any "sun exposure" tag in there because it depends on the season, due to vegetation.

**Comment by rouletout** [comment]
Fair point! So the current porposal is:

- grass
- bushes
- coniferous trees
- deciduous trees

Anything exotic anyone can think of?

**Comment by killakalle** [comment]
I would like to argue that we should also have the tags

- Morning shade
- Noon shade (I think this one is less relevant, but to be consistent with the sunny tags)
- Afternoon shade

They are the exact counterparts to

- Noon sun
- Afternoon sun
- Morning sun

I live in Spain and from April to October you want to know where the shade is to be able to climb. Searching by orientation seems a bit unintuitive to me. Searching for shade seems more straightforward to me.

However, I'm also with @brendanheywood on the more sophisticated model. I've once climbed with a guidebook in Teverga, Spain, that had information about what time the crag was in the sun vs in the shade. And I found this super helpful in planning your day of climbing.

Here's a sample page from Roca Verde:


![image](https://user-images.githubusercontent.com/30682796/83980220-38773d00-a914-11ea-9406-6d078714eff6.png)

PS:  
I posted a related question in the forum a while ago, but there was no response.
https://www.thecrag.com/discussion/2970323994/not-operator-in-filters--search

**Comment by matozoid** [comment]
After editing a lot of crags, I would say there need to be less tags, not more. Maybe make searchable fields for "best time" and "best months" (or better: best part of the day, best season) and leave the rest to the description ("hard to climb in the summer because it is full of nettles")

**Comment by killakalle** [comment]
@matozoid 
To me the important part is not to have additional tags, the important part is to solve the underlying problem.

__How can you search for crags that are suitable for climbing (during a given time)?__

For colder countries, or in the winter season, the _Sunny_ tags solve this issue already. However, for hot countries, e.g. Spain, or the summer season, there is no equivalent solution in place. Additional _Shade_ tags would be one possible solution, though, in my opinion.

>I would say there need to be fewer tags
From a user experience point of view, I find it a bit annoying that I need to scroll through approx. 50 different rock types before I get to the more relevant tags (my personal opinion) - I know you can also type the name of the tag, but I don't know all of them by heart. 

Since we have such a comprehensive list of rock types what about moving them into their own field to avoid cluttering up the list - or move them to the end of the list at least. But that's probably another issue.

**Comment by rouletout** [comment]
@killakalle addresses the right problem - it is not only hard to select the tags (even if you know them), it also discourages correct tagging as it is tedious and not easy to find.
Categorizing it in more than 3 types is one way, but we must re-think th UI for the actual tagging. Ideas like having more groups is one but we could also consider checkboxes (like for project or top rope access) in groups or other easier and more evident ways of choosing the right tags.

We need to make sure tags are added in the first place so that any search function makes sense at all.

**Comment by georg-d** [comment]
+1 for a _searchable_ information on "when climbing is suitable/enjoyable"

**I don't see the combination of vegetation type and rock orientation as sufficient for the purpose.** One example is [Altvogelbach](https://www.thecrag.com/climbing/germany/schwarzwald/area/251600511) in Germany. While it has south oriented sectors starting above the tree shade, the sun does not even touch the crag in winter because sun's anlge is so low it's blocked by the next mountain. Furthermore, the suggested tags would not fit, because "grass" is simply not true, but tagging both tree types (it's a mixed forest) is also misleading for sectors that are suffciently elevated to start above the tree shadow + shelter. 

I'm lucky to join discussion late, so I can stand on the shoulders of giants ;-)  **I support 1) the idea by  @brendanheywood & @killakalle of a start and an end time provided 2) for several month ranges 3) and to take a more general approach like "good/suitable/enjoyable climbing time".** Rationale:
* Sun and shade are relevant, but also legal access restrictions, humidity, tempratures staying high in the evening despite shade coming in, dark or bright rock causing different snow meling speed, etc. => All of this can be condensed into "suitable climbing times" which is the core interest of climbers searching for a crag. The reasons can be detailed in textual descriptions.
* Summer & winter are too limited. Think of Krabi in Thailand which does not have the usual European 4 seasons. Think of mountains causing shadow any month. Think of different vegetation phase timings of different leaf types in different climate. => All of these can be expressed well as month ranges, while more fine grained (exact dates) seem like overkill to me.
* Avoids a considerable extension of the tag list 
* easy to parse by search / filter

I disagree with @lordyavin and @quaestor in the sense **I don't see difficulties to gather the data** because it's usually just 1 time span per crag & "season". This can be _easily_ entered by 1 person for a whole region, and it's realistic to assume sufficient many people will help - especially when being asked for it and if it's convenient - as shown by the huge amount of much more complex time entries in OpenStreetMap. So we may get GUI design inspiration from there, e.g. [BleibtOffen](https://www.bleibtoffen.at/), while a much more simple syntax is sufficient for crags, also a precision of 15...60 minutes is IMHO fine.

**Comment by rouletout** [comment]
We have this proposal for an update to the bulk edit form, mostly re structured tags. It should make it easier to select tags, make it easier useable on mobile and thus support better use / collection of structured tags. Input on the design is welcome:

<img width="1280" alt="1" src="https://user-images.githubusercontent.com/19162952/85948881-eaed5f00-b918-11ea-80d6-ef05677653e3.png">

<img width="1280" alt="2" src="https://user-images.githubusercontent.com/19162952/85948884-efb21300-b918-11ea-8b16-288620fb2c5c.png">

<img width="1280" alt="3" src="https://user-images.githubusercontent.com/19162952/85948885-f3de3080-b918-11ea-9e51-54172efb2541.png">

**Comment by rouletout** [comment]
See also https://github.com/theCrag/website/issues/3742

**Comment by killakalle** [comment]
@rouletout I agree that this looks like an improvement to entering tags.

Though it is not related to the actual issue of this ticket, is it? =)

btw. "Development" is listed twice.

Just to throw in another idea. For the areas where I climb I started creating a Google sheet that provides an overview of sun vs shade plus the time. Here's a screenshot. 

![Screenshot_20200906_171546_com google android apps docs editors sheets](https://user-images.githubusercontent.com/30682796/92329002-e5bd3680-f064-11ea-8880-2534b9709d18.jpg)

I made this sheet mostly for myself, but also because I am in a large WhatsApp group of climbers. And every weekend there is somebody asking the question

- "What sectors have shade in the morning?" or
- "Does anyone know when there is shade at X?" 

It would be really cool if I could add that information to thecrag and generate a sort of overview like this.

**Comment by rouletout** [comment]
@killakalle agree this thread is way to extensive and covers too many topics. We'll split that up. Why this is here is that we were thinking that there are already too many tags to enter for the current form. This improvement will alllow us to provide the tags in a more structured fashion and thus include tags such as vegetation (which was the initial topic here).

Thanks for the hint on development.

May I ask you to add your sheet / proposal to a new issue. I thought there was one covering all sorts of "shade" and "orientation" aspects but i couldn't find it...

**Comment by killakalle** [comment]
@rouletout followed your suggestion and opened a separate ticket (see link above)
